pandas/_libs/lib.pyx

```diff
@@ -1318,7 +1329,11 @@ def infer_dtype(value: object, skipna: bool = True) -> str:
 
     elif PyDateTime_Check(val):
         if is_datetime_array(values):
-            return "datetime"
+            # only check for non nulls to avoid numpy nat
+            if is_datetimetz_array(values[~isnaobj(values)]):
```

Comment:
same

Reply:
Sorry not sure what the problem is here. I will investigate later as it is getting late :sleeping:. If you can tell me a bit more why you think that it would most likely help my research. Thanks!